core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -473,4 +487,27 @@ public static MetricsReporter loadMetricsReporter(Map<String, String> properties
 
     return reporter;
   }
+
+  public static String fullTableName(String catalogName, TableIdentifier identifier) {
```

Comment:
given that this method is being publicly exposed, I think there should be tests that verify different inputs